packages/cli/src/frameworks/nextjs/index.ts

```diff
@@ -89,6 +92,58 @@ export class NextJs implements Framework {
           break;
       }
     }
+
+    const nextJsDir = await detectPagesOrAppDir(path);
+    const nextConfigExists = await detectNextConfigFile(path);
+
+    const require = createRequire(import.meta.url);
+    const nextPath = require.resolve("next");
+    const reactPath = require.resolve("@trigger.dev/react");
+
+    if (nextConfigExists && nextJsDir === "pages" && typeof reactPath !== "undefined") {
+      const nextPackageJsonPath = pathModule.join(nextPath, "../../../", "package.json");
+      const nextPackageJsonData = JSON.parse(await fs.readFile(nextPackageJsonPath, "utf8"));
```

Comment:
It would be much better to use the current way we check for dependencies. 

This function gets the "next" dependency version, and then returns true if it exists. You could create a function that uses this same methodology where you can pass in a package name and it returns undefined or the version string.

https://github.com/triggerdotdev/trigger.dev/blob/6769d6b43930317045796cc31af7924f462df07a/packages/cli/src/frameworks/nextjs/index.ts#L111